File: leetcode questions/21_day_dp/min_cost_climbing_stairs.cpp
#include<iostream>
#include<vector>
using namespace std;
int minCostClimbingStairs(vector<int>& cost){
    int sum=0;

    for (int i = (cost.size()-1); i >= 0; i--){
        if (i==0 || i==1){
            return sum;
        }
        if (cost.at(i)>cost.at(i-1)){
            sum += cost[i-1];
            i=i-1;
        }
        else{
            sum += cost[i];
        }
        
    }
    return sum;
}
int main(){
    vector<int> cost;
    int n;
    cost = {1,100,1,1,1,100,1,1,100,1};
    cout<<minCostClimbingStairs(cost);
}

File: coding platform questions/leetcode questions/daily challenge/july/Min_Cost_Climbing_Stairs.java
public class Min_Cost_Climbing_Stairs {
    public static void main(String[] args) {
        Solution obj = new Solution();
        int[] cost = { 1,100,1,1,1,100,1,1,100,1};
        System.out.println(obj.minCostClimbingStairs(cost));
    }
}
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int prev1= cost[1];
        int prev2= cost[0];
        for (int i = 2; i < cost.length; i++) {
            int temp = prev1;
            prev1 = Math.min(prev1, prev2) + cost[i];
            prev2 = temp;
        }
        return Math.min(prev1,prev2);
    }
}